### PR TITLE
User API Complete, Works on Category Module

### PR DESCRIPTION
**Pull request by niravparmar**
User Module works fine,
in Category module i'm not able to find multiple product like by using id and productTitle , either i can find using id either productTitle, right now i found solution that whatever parameter i get i'll find in both id and title, <---- i'm working on that part is in progress

### CODE REVIEW

File: src/action/categoryAction.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
+const categoryModel = require('../model/category');
+
+module.exports = {
+
+  addCategory: function (req, res) {
+    const {title} = req.body;
+    const categoryData = {title};
+    categoryModel.insert(categoryData).then((data) => {
+      res.json({status: true, data: data, message: 'Add Success'})
```
Comment (suresh-jarvis) [inline src/action/categoryAction.js]: 1) Proper message required in response like : `Category created successfully.`
2) No need use key:val if you have same title Like : res.json({status: true, data, message: 'Add Success'})
3) Have you check query response before send to client ? Like rec. created or not ?

File: src/action/categoryAction.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  getAllCategory: function (req, res) {
+    categoryModel.getAll().then((data) => {
+      res.json({status: true, data: data})
+    })
+  },
+
+  getById: function(req,res) {
+    const {id} = req.params;
+    categoryModel.findById(id).then((data) => {
+      res.json({status:true, data:data})
```
Comment (suresh-jarvis) [inline src/action/categoryAction.js]: What if i pass wrong ID in url which is not exist in DB ?

File: src/action/categoryAction.js [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    })
+  },
+
+  getByTitle : function (req, res) {
+    const { title } = req.params;
+    console.log("request...",req.params);
```
Comment (suresh-jarvis) [inline src/action/categoryAction.js]: Remove console before commit

File: src/action/categoryAction.js [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    categoryModel.findByTitle(title).then((data) => {
+      res.json({status: true, data:data})
+    })
+  },
+
+  delete: function(req,res) {
+    const {id} = req.params;
+    console.log("Id",id);
```
Comment (suresh-jarvis) [inline src/action/categoryAction.js]: Dont use double quote sign in code " use only ' single quote

File: src/action/categoryAction.js [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+    categoryModel.delete(id).then(() => {
+      res.json({status:true, message:'deleted'})
+    })
+  },
+
+  update: function (req, res) {
+    const {id} = req.params;
+    categoryModel.update(id, req.body).then((data) => {
+      res.json({status: true, message: 'Updated', data: data})
```
Comment (suresh-jarvis) [inline src/action/categoryAction.js]: Check query res before send to client

File: src/action/userAction.js [hunk 6]
```diff
@@ -0,0 +1,71 @@
+const UserModel = require('../model/user');
+
+module.exports = {
+
+  signUp: function (req, res) {
+    UserModel.findByEmail(req.body.email).then((data) => {
+      console.log('Data', data);
+      if (Object.keys(data).length > 0) {
+        res.json({status: false, message: 'Email already Exist in the Database!!!'})
```
Comment (suresh-jarvis) [inline src/action/userAction.js]: Dont use signs in message & in the message only 1 char will be caps rest of char should be in lowercase

File: src/model/user.js [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+  isVerified: Boolean
+});
+
+const User = mongoose.model('user', UserSchema);
+
+module.exports = {
+
+  insert: function (userData) {
+    return new Promise((resolve) => {
+      userData['createAt'] = moment().format('YYYY-MM-DD');
```
Comment (suresh-jarvis) [inline src/model/user.js]: Dont use format in createdAt field, System reuires time also so just use moment();

File: src/model/db.js [hunk 8]
```diff
@@ -0,0 +1,15 @@
+const mongoose = require('mongoose');
+
+mongoose.connect('mongodb://localhost/demo');
```
Comment (suresh-jarvis) [inline src/model/db.js]: Use config file for DB connetion string

File: src/action/userAction.js [hunk 9]
```diff
@@ -0,0 +1,71 @@
...
+
+  login: async function (req, res) {
+    const {email, password} = req.body;
+    await UserModel.findByCredential(email, password).then(async (loginRes) => {
+      if (Object.keys(loginRes).length > 0) {
+        console.log('login Response', loginRes);
+        if (!loginRes.isVerified) {
+          res.json({status: false, message: 'Please verify your account!'});
+        } else {
+          res.json({status: true, message: 'Login Successfully', data: loginRes});
```
Comment (suresh-jarvis) [inline src/action/userAction.js]: Dont send password details in response

File: src/route/categoryRoute.js [hunk 10]
```diff
@@ -0,0 +1,12 @@
+const express = require('express');
+const router = express.Router();
+const categoryAction = require('../action/categoryAction');
+
+router.post('/',categoryAction.addCategory);
```
Comment (suresh-jarvis) [inline src/route/categoryRoute.js]: use space after , (comma)

File: src/action/categoryAction.js (hunk 2 above)
Comment (niravparmar) [reply src/action/categoryAction.js]: Tomorrow, i'll add catch where i handle request

File: src/action/categoryAction.js (hunk 3 above)
Comment (niravparmar) [reply src/action/categoryAction.js]: right now the code is not works fine that's why for checking i'll added console for what i get

File: src/action/categoryAction.js (hunk 4 above)
Comment (niravparmar) [reply src/action/categoryAction.js]: okay, gotcha

File: src/action/categoryAction.js (hunk 5 above)
Comment (niravparmar) [reply src/action/categoryAction.js]: okay i will

File: src/action/userAction.js (hunk 6 above)
Comment (niravparmar) [reply src/action/userAction.js]: okay

File: src/action/userAction.js (hunk 9 above)
Comment (niravparmar) [reply src/action/userAction.js]: yeah that's a major bug, gotcha

File: src/route/categoryRoute.js (hunk 10 above)
Comment (niravparmar) [reply src/route/categoryRoute.js]: its a typing mistake ;-)

**Pull request closed without merging**